pkg/csi/manila/controllerserver.go

```diff
@@ -452,24 +452,23 @@ func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 
 	// Try to expand the share
 
-	currentSizeInBytes := int64(share.Size * bytesInGiB)
+	desiredSizeInGiB := bytesToGiB(req.GetCapacityRange().GetRequiredBytes())
 
-	if currentSizeInBytes >= req.GetCapacityRange().GetRequiredBytes() {
+	if share.Size >= desiredSizeInGiB {
```

Comment:
@gman0 trying to understand here , isnt it the same . before you are comparing the size in bytes , now in GB . Is minimum supported size is 1Gi?

Reply:
@ramineni this is actually what causes the bug described in #1764. Here it is comparing values in bytes without first rounding them to GiB. We first need to convert them to GiB and only then check whether the resize should proceed or not.

Without this PR, `req.GetCapacityRange().GetRequiredBytes()` ends up being round down to the closest GiB value. If users use GB units instead of GiB, and the converted values differ only by 1, then `roundDownToGiB(req.GetCapacityRange().GetRequiredBytes()) == currentSizeInGiB`.